backend/src/posts/depends.py

```diff
@@ -31,47 +31,18 @@ def get_article_upvote_details(article_id, user_id, database):
 
 # get news' information according to search_term, and return the data which the function found.
 def get_new_info(search_term, is_initial=False):
-    all_news_data = []
     # iterate pages to get more news data, not actually get all news data
+    # return crawler.get_headline(search_term, (1, 10) if is_initial else 1)
     if is_initial:
-        news = []
-        for page in range(1, 10):
-            pageinfo = {
-                "page": page,
-                "id": f"search:{quote(search_term)}",
-                "channelId": 2,
-                "type": "searchword",
-            }
-            response = requests.get(Constants.NEWS_LINK, params=pageinfo)
-            news.append(response.json()["lists"])
-
-        for result in news:
-            all_news_data.append(result)
+        all_news_data = crawler.startup(search_term=search_term)
     else:
-        pageinfo = {
-            "page": 1,
-            "id": f"search:{quote(search_term)}",
-            "channelId": 2,
-            "type": "searchword",
-        }
-        response = requests.get(Constants.NEWS_LINK, params=pageinfo)
-        all_news_data = response.json()["lists"]
+        all_news_data = crawler.get_headline(search_term,page=Constants.INIT_PAGE_NUM)
     return all_news_data
 
 # add new to database
 def add_new(news_data):
-    session = Session()
-    session.add(NewsArticle(
-        url=news_data["url"],
-        title=news_data["title"],
-        time=news_data["time"],
-        content=" ".join(news_data["content"]),  # 將內容list轉換為字串
-        summary=news_data["summary"],
-        reason=news_data["reason"],
-    ))
-    session.commit()
-    session.close()
-
+    crawler.save(news=news_data)
+    
```

Comment:
It seems like `add_new()` only pass the received parameter `news_data` to `crawler.save(news_data)` without additional process. How about remove the function and just directly use `crawler.save(...)` instead?